pkg/planner/core/point_get_plan.go

```diff
@@ -2071,7 +2071,11 @@ func buildPointDeletePlan(ctx base.PlanContext, pointPlan base.PhysicalPlan, dbN
 	intest.Assert(t != nil, "The point get executor is accessing a table without meta info.")
 	idxs := t.DeletableIndices()
 	deletableCols := t.DeletableCols()
-	colPosInfo, err := buildSingleTableColPosInfoForDelete(pointPlan.OutputNames(), handleCols, idxs, deletableCols, tbl)
+	colPosInfo, err := initColPosInfo(tbl.ID, pointPlan.OutputNames(), handleCols)
+	if err != nil {
+		return nil
+	}
+	_, err = pruneAndBuildSingleTableColPosInfoForDelete(pointPlan.OutputNames(), idxs, deletableCols, tbl, &colPosInfo, 0, nil)
```

Comment:
The https://github.com/pingcap/tidb/blob/master/pkg/planner/optimize.go#L233 fast path plan building should also be considered.

Reply:
the delete op should finally fall into this place. I think it should be safe. I'll add some test